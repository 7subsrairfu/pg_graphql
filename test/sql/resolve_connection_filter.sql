begin;
    create table account(
        id int primary key,
        is_verified bool
    );

    insert into public.account(id, is_verified)
    values
        (1, true),
        (2, true),
        (3, false);

    savepoint a;

    -- Filter by Int
    select jsonb_pretty(
        graphql.resolve($$
            {
              allAccounts(filter: {id: {eq: 2}}) {
                edges {
                  node {
                    id
                  }
                }
              }
            }
        $$)
    );
    rollback to savepoint a;

    -- Filter by Int and bool. has match
    select jsonb_pretty(
        graphql.resolve($$
            {
              allAccounts(filter: {id: {eq: 2}, isVerified: {eq: true}}) {
                edges {
                  node {
                    id
                  }
                }
              }
            }
        $$)
    );
    rollback to savepoint a;

    -- Filter by Int and bool. no match
    select jsonb_pretty(
        graphql.resolve($$
            {
              allAccounts(filter: {id: {eq: 2}, isVerified: {eq: false}}) {
                edges {
                  node {
                    id
                  }
                }
              }
            }
        $$)
    );
    rollback to savepoint a;


    -- Variable: Int
    select jsonb_pretty(
        graphql.resolve($$
           query AccountsFiltered($filt: Int!)
           {
             allAccounts(filter: {id: {eq: $filt}}) {
               edges {
                 node{
                   id
                 }
               }
             }
           }
        $$,
        variables:= '{"filt": 1}'
      )
    );
    rollback to savepoint a;

    -- Variable: IntFilter
    select jsonb_pretty(
        graphql.resolve($$
           query AccountsFiltered($ifilt: IntFilter!)
           {
             allAccounts(filter: {id: $ifilt}) {
               edges {
                 node{
                   id
                 }
               }
             }
           }
        $$,
        variables:= '{"ifilt": {"eq": 3}}'
      )
    );
    rollback to savepoint a;

    -- Variable: AccountFilter
    select jsonb_pretty(
        graphql.resolve($$
           query AccountsFiltered($afilt: AccountFilter!)
           {
             allAccounts(filter: $afilt) {
               edges {
                 node{
                   id
                 }
               }
             }
           }
        $$,
        variables:= '{"ifilt": {"id": {"eq": 1}} }'
      )
    );
    rollback to savepoint a;

rollback;
