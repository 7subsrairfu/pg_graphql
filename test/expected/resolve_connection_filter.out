begin;
    create table account(
        id int primary key,
        is_verified bool
    );
    insert into public.account(id, is_verified)
    values
        (1, true),
        (2, true),
        (3, false);
    savepoint a;
    -- Filter by Int
    select jsonb_pretty(
        graphql.resolve($$
            {
              allAccounts(filter: {id: {eq: 2}}) {
                edges {
                  node {
                    id
                  }
                }
              }
            }
        $$)
    );
          jsonb_pretty           
---------------------------------
 {                              +
     "data": {                  +
         "allAccounts": {       +
             "edges": [         +
                 {              +
                     "node": {  +
                         "id": 2+
                     }          +
                 }              +
             ]                  +
         }                      +
     },                         +
     "errors": [                +
     ]                          +
 }
(1 row)

    rollback to savepoint a;
    -- Filter by Int and bool. has match
    select jsonb_pretty(
        graphql.resolve($$
            {
              allAccounts(filter: {id: {eq: 2}, isVerified: {eq: true}}) {
                edges {
                  node {
                    id
                  }
                }
              }
            }
        $$)
    );
          jsonb_pretty           
---------------------------------
 {                              +
     "data": {                  +
         "allAccounts": {       +
             "edges": [         +
                 {              +
                     "node": {  +
                         "id": 2+
                     }          +
                 }              +
             ]                  +
         }                      +
     },                         +
     "errors": [                +
     ]                          +
 }
(1 row)

    rollback to savepoint a;
    -- Filter by Int and bool. no match
    select jsonb_pretty(
        graphql.resolve($$
            {
              allAccounts(filter: {id: {eq: 2}, isVerified: {eq: false}}) {
                edges {
                  node {
                    id
                  }
                }
              }
            }
        $$)
    );
       jsonb_pretty       
--------------------------
 {                       +
     "data": {           +
         "allAccounts": {+
             "edges": [  +
             ]           +
         }               +
     },                  +
     "errors": [         +
     ]                   +
 }
(1 row)

    rollback to savepoint a;
    -- Variable: Int
    select jsonb_pretty(
        graphql.resolve($$
           query AccountsFiltered($filt: Int!)
           {
             allAccounts(filter: {id: {eq: $filt}}) {
               edges {
                 node{
                   id
                 }
               }
             }
           }
        $$,
        variables:= '{"filt": 1}'
      )
    );
          jsonb_pretty           
---------------------------------
 {                              +
     "data": {                  +
         "allAccounts": {       +
             "edges": [         +
                 {              +
                     "node": {  +
                         "id": 1+
                     }          +
                 }              +
             ]                  +
         }                      +
     },                         +
     "errors": [                +
     ]                          +
 }
(1 row)

    rollback to savepoint a;
    -- Variable: IntFilter
    select jsonb_pretty(
        graphql.resolve($$
           query AccountsFiltered($ifilt: IntFilter!)
           {
             allAccounts(filter: {id: $ifilt}) {
               edges {
                 node{
                   id
                 }
               }
             }
           }
        $$,
        variables:= '{"ifilt": {"eq": 3}}'
      )
    );
          jsonb_pretty           
---------------------------------
 {                              +
     "data": {                  +
         "allAccounts": {       +
             "edges": [         +
                 {              +
                     "node": {  +
                         "id": 3+
                     }          +
                 }              +
             ]                  +
         }                      +
     },                         +
     "errors": [                +
     ]                          +
 }
(1 row)

    rollback to savepoint a;
    -- Variable: AccountFilter
    select jsonb_pretty(
        graphql.resolve($$
           query AccountsFiltered($afilt: AccountFilter!)
           {
             allAccounts(filter: $afilt) {
               edges {
                 node{
                   id
                 }
               }
             }
           }
        $$,
        variables:= '{"ifilt": {"id": {"eq": 1}} }'
      )
    );
                 jsonb_pretty                  
-----------------------------------------------
 {                                            +
     "data": null,                            +
     "errors": [                              +
         "Variable filter not (yet) supported"+
     ]                                        +
 }
(1 row)

    rollback to savepoint a;
rollback;
